source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -183,6 +183,13 @@ void SslSocket::resumeHandshake() {
 Network::Connection& SslSocket::connection() const { return callbacks_->connection(); }
 
 void SslSocket::onSuccess(SSL* ssl) {
+  if (SSL_is_server(ssl) && SSL_session_reused(ssl) &&
+      SSL_get_verify_mode(ssl) != SSL_VERIFY_NONE) {
+    // SSL_CTX_set_reverify_on_resume() is not respected for servers. Invoke the verify callback
+    // directly here in order to populate the certificate validation status.
+    uint8_t unused_alert;
+    ContextImpl::customVerifyCallback(ssl, &unused_alert);
```

Comment:
Potentially problematic cases:

* What if this returns `ssl_verify_retry`, which means that an async verification was started?
* If this returns `ssl_verify_invalid`, should the connection be killed? Is this even possible (given that it didn't happen when the session was first handshaked)?

Reply:
@lizan @danzh2010 @RyanTheOptimist Any other feedback on issues with invoking the verify callback here?